raytracing/matrix.py

```diff
@@ -136,6 +136,10 @@ def __init__(
         self.isFlipped = False
         super(Matrix, self).__init__()
 
+    @property
+    def isIdentity(self):
+        return self.A == 1 and self.D == 1 and self.B == 0 and self.C == 0
```

Comment:
Understand the reasoning: if it is identify, then we should do that.  But I think the source of all that is that the index is really "undetermined" more than anything.  I wonder what would happen if we set index to None by default?

Reply:
It is something possible, but it requires a lot of changes, everywhere: in the doc, we assume a lot of times that n = 1 by default, so we have to change it. `Lens`, `CurvedMirror` assume that n = 1 by default.